torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2990,6 +2990,117 @@ def generator():
 
     return list(generator())
 
+def sample_inputs_max_unpool1d(op_info, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    kerneli = [(1,), 3]
+    stridei = [None, (2,)]
+    Ni = [1]
+    Ci = [2]
+    Wi = [6]
+    ceil_modei = [True, False]
+    paddingi = [0]
+    dilationi = [1]
+    output_sizei = [True, False]
+
+    products = product(kerneli, stridei, Ni, Ci, Wi, ceil_modei, paddingi, dilationi, output_sizei)
+
+    def generator():
+        for kernel_size, stride, N, C, W, ceil_mode, padding, dilation, output_size in products:
+            input_tensor = make_arg((N, C, W))
+
+            max_unpool1d = torch.nn.MaxUnpool1d(kernel_size, stride, padding)
+            max_pool1d = torch.nn.MaxPool1d(kernel_size, stride, ceil_mode=ceil_mode, padding=padding,
+                                            dilation=dilation, return_indices=True)
+
+            kwargs = {
+                "kernel_size": max_unpool1d.kernel_size,
+                "stride": max_unpool1d.stride,
+                "padding": max_unpool1d.padding,
+            }
+
+            output_pool, indices = max_pool1d(input_tensor)
+            if output_size:
+                kwargs.update({'output_size': input_tensor.size()})
+            yield SampleInput(output_pool.detach().requires_grad_(requires_grad), args=(indices,), kwargs=kwargs)
```

Comment:
This `output_pool.detach().requires_grad_(requires_grad)` is to make sure that there is no `grad_fn` history recorded for `output_pool` tensor. If not done, this will fail the `test_noncontiguous_samples` test:

https://github.com/pytorch/pytorch/blob/06d1be2447ad2041336ee93c435345a4f371b9a9/test/test_ops.py#L228

Good to have that assertion in the test, else I would have unknowingly forgotten to do this.

cc: @mruberry